sale_order_line_date/models/sale_order.py

```diff
@@ -0,0 +1,30 @@
+# © 2016 OdooMRP team
+# © 2016 AvanzOSC
+# © 2016 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# © 2016 Eficent Business and IT Consulting Services, S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    @api.onchange('commitment_date')
+    def _onchange_commitment_date(self):
+        """Warn if the commitment dates is sooner than the commitment date"""
+        result = super(SaleOrder, self)._onchange_commitment_date()
+        if not result:
+            result = {}
+        if not self:
+            return result
+        self.ensure_one()
+        if 'warning' not in result:
+            lines = []
+            for line in self.order_line:
+                lines.append((1, line.id, {'commitment_date':
```

Comment:
```suggestion
                if not line.commitment_date:
                    lines.append((1, line.id, {'commitment_date':
```
Is there a reason why the order commitment date should overwrite an existing line commitment date? 

If not, there is a solution to the issue with the date not propagating to picking. The reason the picking scheduled date doesn't inherit properly is because it's looking for an order commitment date. If you add an onchange to the line commitment date that overwrites order commitment date if it doesn't exist, then the picking will have a date to inherit.